src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,547 @@
+  CborError err = CborNoError;
+  err = cbor_encode_double(&sample_array, local_sample_member);
+  return err;
+}
+
+/**
+ * @brief Adds a new item to the report builder queue.
+ *
+ * This function creates a new report builder queue item and adds it to the queue. The item's properties are set based on the message type.
+ *
+ * @param node_id The ID of the node for the report.
+ * @param sensor_type The type of the sensor for the report.
+ * @param sensor_data Pointer to the sensor data for the report.
+ * @param sensor_data_size The size of the sensor data.
+ * @param msg_type The type of the message to be added to the queue.
+ *
+ * @note If the message type is REPORT_BUILDER_SAMPLE_MESSAGE, the node_id, sensor_type, sensor_data, and sensor_data_size parameters are used.
+ * If the message type is REPORT_BUILDER_INCREMENT_SAMPLE_COUNT or REPORT_BUILDER_CHECK_CRC, these parameters are ignored and default values are used instead
+ * since they are not needed for these message types.
+ */
+void reportBuilderAddToQueue(uint64_t node_id, uint8_t sensor_type, void *sensor_data, uint32_t sensor_data_size, report_builder_message_e msg_type) {
+  report_builder_queue_item_t item;
+  item.message_type = msg_type;
+  if (msg_type == REPORT_BUILDER_SAMPLE_MESSAGE) {
+    item.node_id = node_id;
+    item.sensor_type = sensor_type;
+    // This will be freed by the task after the task is done using the sensor_data
+    item.sensor_data = static_cast<void *>(pvPortMalloc(sensor_data_size));
+    configASSERT(item.sensor_data != NULL);
+    memcpy(item.sensor_data, sensor_data, sensor_data_size);
+    item.sensor_data_size = sensor_data_size;
+  } else if (msg_type == REPORT_BUILDER_INCREMENT_SAMPLE_COUNT || msg_type == REPORT_BUILDER_CHECK_CRC) {
+    item.node_id = 0;
+    item.sensor_type = 0;
+    item.sensor_data = NULL;
+    item.sensor_data_size = 0;
+  } else {
+    configASSERT(0);
+  }
+  xQueueSend(_report_builder_queue, &item, portMAX_DELAY);
+}
+
+/**
+ * @brief Adds samples to the sensor report.
+ *
+ * This function takes the sensor data and adds the samples to the sensor report. The samples are added based on the sensor type and the number of samples.
+ *
+ * @param context The context for the sensor report encoder.
+ * @param sensor_type The type of the sensor for the report.
+ * @param sensor_data Pointer to the sensor data for the report.
+ * @param sample_index The index of the sensor data to be copied into the sensor report.
+ */
+static bool addSamplesToReport(sensor_report_encoder_context_t &context, uint8_t sensor_type, void *sensor_data, uint32_t sample_index) {
+  bool rval = false;
+  switch (sensor_type){
+    case AANDERAA_SENSOR_TYPE: {
+      aanderaa_aggregations_t aanderaa_sample = (static_cast<aanderaa_aggregations_t *>(sensor_data))[sample_index];
+      if (sensor_report_encoder_open_sample(context, AANDERAA_NUM_SAMPLE_MEMBERS) != CborNoError) {
+        BRIDGE_LOG_PRINT("Failed to open sample in addSamplesToReport\n");
```

Comment:
I think we need to break in these statements if the encoding fails.

Reply:
good catch yup!